x-pack/plugins/ml/server/shared_services/providers/anomaly_detectors.ts

```diff
@@ -23,7 +23,13 @@ export function getAnomalyDetectorsProvider({
 }: SharedServicesChecks): AnomalyDetectorsProvider {
   return {
     anomalyDetectorsProvider(mlClusterClient: ILegacyScopedClusterClient, request: KibanaRequest) {
-      const hasMlCapabilities = getHasMlCapabilities(request);
+      //  APM is using this service in anomaly alert, kibana alerting doesn't provide request object
+      // So we are adding a dummy request for now
+      // TODO: Remove this once kibana alerting provides request object
+      const hasMlCapabilities =
+        request.params !== 'DummyKibanaRequest'
+          ? getHasMlCapabilities(request)
+          : (_caps: string[]) => Promise.resolve();
```

Comment:
Okay, I see now why you used "DummyKibanaRequest". This doesn't seem like the right approach. Either `request` should be optional or we should not use `anomalyDetectorsProvider`. Passing in a dummy request object is going to cause confusion down the line I think.

Reply:
`DummyKibanaRequest` will not be needed after this PR is merged https://github.com/elastic/kibana/pull/74965
However a fake request object will still be needed until https://github.com/elastic/kibana/issues/64588 is fixed.
Making `request` optional will require more changes to undo our work arounds once this issue is resolved.

Reply:
@jgowdyelastic do i need to hold off on merging this PR until yours is merged first? LMK if i should make any changes to the `DummyKibanaRequest` in the meantime.

Reply:
that PR is a few days away from being merged. so i'm happy incorporate and update the changes in this PR if you merge first.